packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -336,7 +336,7 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
         const storagePath = configStorage.hostStoragePath ? join(configStorage.hostStoragePath, plugin.model.id) : undefined;
         const globalStoragePath = join(configStorage.hostGlobalStoragePath, plugin.model.id);
         const pluginContext: theia.PluginContext = {
-            extensionPath: plugin.pluginFolder,
+            extensionPath: plugin.pluginFolder.replace(/\/$/, ""),
```

Comment:
looking at the code i am not sure how we can get such paths, do you know how to reproduce? Otherwise I'm not sure that it is a correct place to fix or whether it has to be fixed in Theia at all

Reply:
I'm not sure I understand your question properly.
To test it you can generate `HellowWorld` extension and add the code to print `extensionPath` to output.
```
   const output = vscode.window.createOutputChannel('HelloWorld');
   output.appendLine(`Extension Path ${context.extensionPath}`);
   output.show(true);
```

I launched helloworld extension in VS Code (see the output)
![Screenshot from 2020-06-18 15-54-39](https://user-images.githubusercontent.com/1655894/85032347-65371a00-b188-11ea-9c6e-9f412220078d.png)

and in Che-Theia
![Screenshot from 2020-06-18 17-14-54](https://user-images.githubusercontent.com/1655894/85032373-6d8f5500-b188-11ea-9054-1a861c499656.png)

Believe me we don't do anything with `extensionPath` in Che-Theia, it comes from `plugin-ext` package.

Why do I remove the slash when setting `extensionPath`?
I see `pluginFolder` is used in many places, I prefer to leave it unchanged.

Reply:
> I see pluginFolder is used in many places, I prefer to leave it unchanged.

There should not be slashes at the end of folder paths. We should fix the place where path is produced.

Reply:
That's the place: https://github.com/eclipse-theia/theia/blob/d9d6e7092377c96e91dcbab6c3ebf7533aacd250/packages/plugin-ext/src/hosted/node/plugin-reader.ts#L95

It was introduce in the initial plugin system PR when the code was not using Node or Theia APis to manipulate paths. We can try remove it now. 

We generally should get rid of passing fs paths as part of remote protocols, since it violates paths portability: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#uri-over-path `PluginPackage.packagePath` and `PluginModel.packagePath` should be replaced with `packageUri` which is constructed via `URI` class. In `PluginModel.packagePath` is already deprecated btw.

Reply:
@akosyakov it looks it's even easier than I thought. I just removed `pluginPath = path.normalize(pluginPath + '/'); ` from `plugin-reader.ts` and it became working.
I also checked the plugins to be working. Seems everything's ok.

Reply:
I could find only one place where we don't use yet `path.resolve` or `path.join` Node.js api: https://github.com/eclipse-theia/theia/blob/59226ddabeeaab2df198bfd85e9f7a2d678cffa6/packages/plugin-ext/src/hosted/node-electron/scanner-theia-electron.ts#L25 Could you update it please to use one of them?

Reply:
Done